### ISSUE TITLE
[Chromium] Blocked iframes not always collapsed on stackoverflow.com

### ISSUE THREAD

**Issue by gorhill** [uBlock-LLC/uBlock#673]
Any question page.

From what I've gathered so far, a proper `load` or `error` event is emitted by the frame, however, the `src` attribute is empty -- thus the URL can't be checked against the filters.

Need to investigate further.

**Comment by harshanvn** [comment]
another test case (disabled dynamic filtering as seen in the screenshot) , having the similar issue - vr-zone.com
![ublock vrzone](https://cloud.githubusercontent.com/assets/779630/6025497/eeb9fcb6-abf8-11e4-9b51-85c52ac8facb.png)

**Comment by gorhill** [comment]
I don't see any `iframe` in there. Looks to me another issue, assuming there is any. Let's keep all issues separated, open another one please if you think it's the same. I prefer to resolve as dup than to fill an issue with unrelated stuff, unless investigation confirm it's the same issue.

**Comment by my-password-is-password** [comment]
![ublockstackoverflowadzerk](https://cloud.githubusercontent.com/assets/6408935/6048770/95c6c9e4-ac67-11e4-88ac-99fd0ec6cdb2.png)

`adzerk.net/ados.js` is responsible for the iframe ads and If you're using uBlock's default filters, EasyList has `||adzerk.net^$third-party` but it also has 3 exceptions.

```
@@||engine.adzerk.net/ados?$script,domain=stackoverflow.com
@@||static.adzerk.net/ados.js$domain=stackoverflow.com
@@||adzerk.net/ados.js$domain=majorleaguegaming.com
```

**Comment by gorhill** [comment]
Well whatever EasyList does, I get this:

![a](https://cloud.githubusercontent.com/assets/585534/6049288/1ce83b58-ac84-11e4-80bb-9ebe895ebf1b.png)

Bottom-line, the content of a frame was blocked, uBlock is supposed to collapse it. It's kind of random though, probably because the blocked content collapsing code is split in two parts, the pre- and post-DOM loaded event.

I don't know why there are exceptions in EasyList for `adzerk.net` on StackOverflow -- but that's not the issue here. It looks to me the site works fine without that exception. Report to EasyLIst if you want, I just don't have the time.

**Comment by gorhill** [comment]
Got curious as to why part of `adzerk.net` was whitelisted: https://hg.adblockplus.org/easylist/rev/d2083c34a335

**Comment by chrisaljoudi** [comment]
@gorhill while reproducing this, it appears that the `sub_frame` request for the `iframe` is blocked indeed. However, the `iframe` has an explicit `width` and `height` set, and it doesn't appear that µBlock does anything to explicitly remove/hide elements for which it blocks the requests (when it makes sense to).

Which is to say: `iframe`, `img`, `object`, `video`, `audio`.

Am I missing something?

**Comment by gorhill** [comment]
> it doesn't appear that µBlock does anything to explicitly remove/hide elements for which it blocks the requests

Of course it does. The problem here is that the `src` property is empty, instead of being set with the URL of the blocked request.

**Comment by chrisaljoudi** [comment]
@gorhill right, that's what I'd thought, but I couldn't trace it along the chain of a `sub_frame` request through µBlock.

I'm sorry for the inconvenience, but mind pointing me to the part that removes/hides the blocked element from the DOM? 

I understand that a cosmetic filter might match the element, but it also seems like it's a good idea to manually set `.style.display = "none";` for targets of certain types of requests (`sub_frame`, `img`, etc.)

**Comment by gorhill** [comment]
May be related to #618.

**Comment by gorhill** [comment]
Appears fixed with fix to #886. In both cases, it involved frame in a frame.
